File: TimeTrackerWeb/src/main/java/de/mg/tt/rest/CORSResponseFilter.java
package de.mg.tt.rest;


import javax.servlet.*;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;

public class CORSResponseFilter implements Filter {

    @Override
    public void init(FilterConfig filterConfig) throws ServletException {

    }

    @Override
    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
        if (response instanceof HttpServletResponse) {
            HttpServletResponse http = (HttpServletResponse) response;
            http.addHeader("Access-Control-Allow-Origin", "*");
            http.addHeader("Access-Control-Allow-Credentials", "true");
            http.addHeader("Access-Control-Allow-Methods", "GET, POST, DELETE, PUT");
            http.addHeader("Access-Control-Allow-Headers", "content-type, accept");
        }
        chain.doFilter(request, response);
    }

    @Override
    public void destroy() {

    }


}

File: TimeTrackerWeb/src/main/scala/de/mg/tt/rest/CORSResponseFilter.scala
package de.mg.tt.rest

import java.io.IOException

import javax.servlet._
import javax.servlet.http.HttpServletResponse

class CORSResponseFilter extends Filter {

  @throws[ServletException]
  override def init(filterConfig: FilterConfig): Unit = {
  }

  @throws[IOException]
  @throws[ServletException]
  override def doFilter(request: ServletRequest, response: ServletResponse, chain: FilterChain): Unit = {
    response match {
      case http: HttpServletResponse =>
        http.addHeader("Access-Control-Allow-Origin", "*")
        http.addHeader("Access-Control-Allow-Credentials", "true")
        http.addHeader("Access-Control-Allow-Methods", "GET, POST, DELETE, PUT")
        http.addHeader("Access-Control-Allow-Headers", "content-type, accept, Authorization")
      case _ =>
    }
    chain.doFilter(request, response)
  }

  override def destroy(): Unit = {
  }
}